src/feature/dirauth/voteflags.c

```diff
@@ -608,6 +609,11 @@ set_routerstatus_from_routerinfo(routerstatus_t *rs,
   rs->dir_port = ri->dir_port;
   rs->is_v2_dir = ri->supports_tunnelled_dir_requests;
 
+  if (ri->cache_info.signing_key_cert) {
+    ed25519_pubkey_copy(&rs->ed25519_signing_key,
```

Comment:
This code leaves the key zeroed-out if the relay does not have an ed25529 key.
Instead, allocate memory for the key if it is non-NULL.
And free that memory in routerstatus_free_().

Reply:
0d36d1f514e308d741f2ba232276115458aa86f8